src/types.ts

```diff
@@ -0,0 +1,148 @@
+import Client from "./client";
```

Comment:
In general I think it's better to have types defined as close as possible to where they are used in public declarations.
Not sure if it's necessary to have them all in a single file, but it's not that big of a deal, since we don't have many anyways.